### PR TITLE
[4.3] Guided Tours fix DisplayController

### PR DESCRIPTION
**Pull request by heelc29**
### Summary of Changes

fix component name `com_guidedtours` in DisplayController
~~convert 2nd if to elseif (since the conditions are mutually exclusive)~~

### Testing Instructions

open `/administrator/index.php?option=com_guidedtours&view=tour&layout=edit&id=1`
open `/administrator/index.php?option=com_guidedtours&view=step&layout=edit&id=1`

### Actual result BEFORE applying this Pull Request

`404 Component not found`

image

### Expected result AFTER applying this Pull Request

image

### Link to documentations
Please select:

- [x] No documentation changes for docs.joomla.org needed
- [x] No documentation changes for manual.joomla.org needed

### CODE REVIEW

**Comment by richard67** [comment]
I have tested this item :white_check_mark: successfully on 5a5a1fe84eefa43cb29baf3801551ffa3a774749<br /><br />Hint for other testers: It needs to enter the URL directly in the URL field of the browser to reproduce the issue and test the fix.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/40320">issues.joomla.org/tracker/joomla-cms/40320</a>.</sub>

File: administrator/components/com_guidedtours/src/Controller/DisplayController.php [hunk 1]
```diff
@@ -51,14 +51,12 @@ public function display($cachable = false, $urlparams = [])
...
         if ($view === 'tour' && $layout === 'edit' && !$this->checkEditId('com_guidedtours.edit.tour', $id)) {
             $this->setMessage(Text::sprintf('JLIB_APPLICATION_ERROR_UNHELD_ID', $id), 'error');
-            $this->setRedirect(Route::_('index.php?option=com_guidedtour&view=tours', false));
+            $this->setRedirect(Route::_('index.php?option=com_guidedtours&view=tours', false));
 
             return false;
-        }
-
-        if ($view === 'step' && $layout === 'edit' && !$this->checkEditId('com_guidedtours.edit.step', $id)) {
+        } elseif ($view === 'step' && $layout === 'edit' && !$this->checkEditId('com_guidedtours.edit.step', $id)) {
```
Comment (richard67) [inline administrator/components/com_guidedtours/src/Controller/DisplayController.php]: I'm not sure if this change is needed as in the previous `if` condition there is a `return` at the end, so the condition here will be anyway only checked when the previous one was false.

Comment (richard67) [reply administrator/components/com_guidedtours/src/Controller/DisplayController.php]: Regarding the early return pattern, the previous code (i.e. without the change from this PR) was more suitable.

Comment (heelc29) [reply administrator/components/com_guidedtours/src/Controller/DisplayController.php]: Oriented on com_user. Should I revert it?
https://github.com/joomla/joomla-cms/blob/4.3-dev/administrator/components/com_users/src/Controller/DisplayController.php#L86-L128

Comment (richard67) [reply administrator/components/com_guidedtours/src/Controller/DisplayController.php]: I'd prefer yes, revert.

Comment (heelc29) [inline administrator/components/com_guidedtours/src/Controller/DisplayController.php]:
```suggestion
        }

        if ($view === 'step' && $layout === 'edit' && !$this->checkEditId('com_guidedtours.edit.step', $id)) {
```

**Review by heelc29 — COMMENTED** [review]
revert

File: administrator/components/com_guidedtours/src/Controller/DisplayController.php (hunk 1 above)
Comment (heelc29) [reply administrator/components/com_guidedtours/src/Controller/DisplayController.php]: Done

**Comment by richard67** [comment]
I have tested this item :white_check_mark: successfully on e2f9efb2d36a9e26f84847428b614c34a81a477d<br /><br />Hint for other testers: It needs to enter the URL directly in the URL field of the browser to reproduce the issue and test the fix.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/40320">issues.joomla.org/tracker/joomla-cms/40320</a>.</sub>

**Comment by Quy** [comment]
I have tested this item :white_check_mark: successfully on e2f9efb2d36a9e26f84847428b614c34a81a477d<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/40320">issues.joomla.org/tracker/joomla-cms/40320</a>.</sub>

**Comment by Quy** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/40320">issues.joomla.org/tracker/joomla-cms/40320</a>.</sub>

**Pull request merged**